contracts/thecosomata/ThecosomataETH.sol

```diff
@@ -0,0 +1,219 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity ^0.8.0;
+
+import {AccessControl} from "@openzeppelin/contracts/access/AccessControl.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+interface IBTRFLY is IERC20 {
```

Comment:
Just looked at IERC20.sol and surprised that it doesn't have the burn and decimals methods (not to mention others).

Can we rename this `IERC20Extended` since we're using it once for eth?